modules/developers-guide/pages/authnz.adoc

```diff
@@ -0,0 +1,502 @@
+----
+# create role for demos
+CREATE ROLE IF NOT EXISTS 'web_user' WITH PASSWORD = 'web_user' AND LOGIN = TRUE;
+
+# create REST API example
+CREATE KEYSPACE IF NOT EXISTS store WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':'1'};
+
+CREATE TABLE IF NOT EXISTS store.shopping_cart (userid text PRIMARY KEY, item_count int, last_update_timestamp timestamp);
+
+INSERT INTO store.shopping_cart (userid, item_count, last_update_timestamp) VALUES ('9876', 2, toTimeStamp(toDate(now())));
+INSERT INTO store.shopping_cart (userid, item_count, last_update_timestamp) VALUES ('1234', 5, toTimeStamp(toDate(now())));
+
+GRANT MODIFY ON TABLE store.shopping_cart TO web_user;
+GRANT SELECT ON TABLE store.shopping_cart TO web_user;
+
+# create GraphQL example
+CREATE KEYSPACE IF NOT EXISTS library WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':'1'};
+
+CREATE TABLE IF NOT EXISTS library.books (title text PRIMARY KEY, author text);
+CREATE TABLE IF NOT EXISTS library.authors (name text PRIMARY KEY, title text);
+
+GRANT MODIFY ON TABLE library.books TO web_user;
+GRANT MODIFY ON TABLE library.authors TO web_user;
+GRANT SELECT ON TABLE library.books TO web_user;
+GRANT SELECT ON TABLE library.authors TO web_user;
+----
+
+=== Create a keycloak user
+
+Create a keycloak user, so that the user's JWT can be used to verify authentication and authorization later when a query is made.
+
+[source, bash]
+----
+TOKEN=$(curl -s --data "username=admin&password=admin&grant_type=password&client_id=admin-cli" http://localhost:4444/auth/realms/master/protocol/openid-connect/token | jq -r '.access_token') //<1>
+
+curl -L -X POST 'http://localhost:4444/auth/admin/realms/stargate/users' \
+-H "Content-Type: application/json" \
+-H "Authorization: bearer $TOKEN" \ //<2>
+--data-raw '{
+    "username": "testuser1", //<3>
+    "enabled": true,
+    "emailVerified": true,
+    "attributes": {
+        "userid": [
+            "9876" //<4>
+        ],
+        "role": [
+            "web_user" //<5>
+        ]
+    },
+    "credentials": [
+        {
+            "type": "password",
+            "value": "testuser1", //<6>
+            "temporary": "false"
+        }
+    ]
+}'
+----
+<1> Generate a token to access keycloak. The keycloak administrator username and password are used in this line.
+<2> The token generated in the first line is used to authorize the REST query access to keycloak to add a user.
+<3> The keycloak username is identified.
+<4> The userid attribute used in the Cassandra table column to verify accessibility is identified.
+<5> The Cassandra role is identified.
+<6> The keycloak password is identified.
+
+This sequence of commands access Keycloak as an administrator, to create the information required to
+validate a user and grant access.
+
+== Test the auth/auth
+
+Now let's try some queries to see if the auth/auth is working.
+
+This command should work, because the `userid` of the `testuser1` user has the role
```

Comment:
Because it has select and because it has the appropriate claim `x-stargate-userid` 

Would it help to show the decoded jwt? You can decode with jwt.io